src/components/Editor/AddProblemModal.tsx

```diff
@@ -33,96 +23,64 @@ async function addProblem(
-              enterFrom="opacity-0 scale-95"
-              enterTo="opacity-100 scale-100"
-              leave="ease-in duration-200"
-              leaveFrom="opacity-100 scale-100"
-              leaveTo="opacity-0 scale-95"
-            >
-              <Dialog.Panel className="w-full max-w-md transform overflow-hidden rounded-2xl bg-black text-white p-6 text-left align-middle shadow-xl transition-all">
-                <Dialog.Title as="h3" className="text-lg font-medium leading-6">
-                  Add Problem
-                </Dialog.Title>
-                <div className="mt-2 relative rounded-md shadow-sm">
-                  <input
-                    type="text"
-                    className="shadow-sm focus:ring-blue-500 focus:border-blue-500 block w-full sm:text-sm border-gray-300 rounded-md dark:bg-gray-900 dark:border-gray-700"
-                    placeholder="Enter Problem URL"
-                    onChange={e => console.log(e.target.value)}
-                    ref={linkRef}
-                  />
-                </div>
+    <Modal {...props}>
+      <Dialog.Panel className="w-full max-w-2xl transform overflow-hidden rounded-2xl bg-black text-white p-6 text-left align-middle shadow-xl transition-all">
+        <Dialog.Title as="h3" className="text-lg font-medium leading-6">
+          Add Problem
+        </Dialog.Title>
+        <div className="mt-2 relative rounded-md shadow-sm">
+          <input
+            type="text"
+            className="shadow-sm focus:ring-blue-500 focus:border-blue-500 block w-full sm:text-sm border-gray-300 rounded-md dark:bg-gray-900 dark:border-gray-700"
+            placeholder="Enter Problem URL"
+            onChange={e => console.log(e.target.value)}
+            ref={linkRef}
+          />
+        </div>
 
-                <div className="mt-4">
-                  <button
-                    className="btn"
-                    onClick={() =>
-                      linkRef.current &&
-                      addProblem(linkRef.current.value, setMetadata, setStatus)
-                    }
-                  >
-                    {status}
-                  </button>
-                </div>
-                <div className="mt-4 relative">
-                  <pre className="bg-gray-900 p-4 rounded-md text-white text-xs whitespace-pre-wrap">
-                    {metadata}
-                  </pre>
-                  <CopyButton
-                    className="btn absolute top-2 right-2"
-                    onClick={() => {
-                      navigator.clipboard.writeText(metadata);
-                    }}
-                  />
-                </div>
-              </Dialog.Panel>
-            </Transition.Child>
-          </div>
+        <div className="mt-4">
+          <button
+            className="btn"
+            onClick={() =>
+              linkRef.current &&
+              addProblem(linkRef.current.value, setMetadata, setStatus)
+            }
+          >
```

Comment:
should probably add `disabled={status!=...}` here too, otherwise if someone clicks the button twice `addProblem` might get run twice